app/services/sso_service.rb

```diff
@@ -33,17 +33,18 @@ def initialize(response)
   validate :composite_validations
 
   def persist_authentication!
-    if new_login?
-      # FIXME: possibly revisit this. Is there a possibility that different sign-in contexts could get
-      # merged? MHV LOA1 -> IDME LOA3 is ok, DS Logon LOA1 -> IDME LOA3 is ok, everything else is not.
-      # because user, session, user_identity all have the same TTL, this is probably not a problem.
-      mergable_identity_attributes.each do |attribute|
-        new_user_identity.send(attribute + '=', existing_user.identity.send(attribute))
-      end
-      existing_user.destroy
-    end
+    existing_user.destroy if new_login?
 
     if valid?
+      if new_login?
```

Comment:
@saneshark does line 36 make this unreachable code?

Reply:
no, because destroying the user, just detroys it in redis, existing_user object should still exist in memory, just frozen.